sktime/forecasting/compose/tests/test_pipeline.py

```diff
@@ -76,3 +75,34 @@ def test_skip_inverse_transform():
     forecaster.fit(y_train, fh=fh)
     y_pred = forecaster.predict()
     assert isinstance(y_pred, pd.Series)
+
+
+def test_nesting_pipelines():
+    """Test that nesting of pipelines works."""
+    from sktime.forecasting.ets import AutoETS
+    from sktime.transformations.series.boxcox import LogTransformer
+    from sktime.transformations.series.compose import OptionalPassthrough
+    from sktime.transformations.series.detrend import Detrender
+    from sktime.utils._testing.scenarios_forecasting import (
+        ForecasterFitPredictUnivariateWithX,
+    )
+
+    pipe = ForecastingPipeline(
+        steps=[
+            ("logX", OptionalPassthrough(LogTransformer())),
+            ("detrenderX", OptionalPassthrough(Detrender(forecaster=AutoETS()))),
+            (
+                "etsforecaster",
+                TransformedTargetForecaster(
+                    steps=[
+                        ("log", OptionalPassthrough(LogTransformer())),
+                        ("autoETS", AutoETS()),
```

Comment:
I think the `OptionalPassthrough` is rudimentary here, so would remove it

Reply:
I wanted composites inside the pipeline, to make sure that nesting works

Reply:
in this case, more nesting is better, even if it overcomplicates the pipeline